docs/Model Pipeline.md

```diff
@@ -0,0 +1,351 @@
+- Write programs to work together.
+- Write programs to handle text streams, because that is a universal interface.
+
+Our approach to models in the [Impact Engine Framework](Impact%20Engine%20Framework.md) is the same. 
+- Each model will do one thing and do it well. 
+- Models work with each other. 
+- The [Impl (Impact YAML)](Impl%20(Impact%20YAML).md) text format is the communication medium between models.
+
+Calculating a component's impacts often requires using multiple models in sequence. Each model takes as input the outputs of the previous model in the chain, all working together to calculate impacts from observations.
+
+```mermaid 
+flowchart LR
+Observations --> Model1 --> Model2 --> Model3 --> Impacts
+```
+
+At the start of the chain, we input source observations. Each model in the chain takes as input the observations and modifies them somehow before passing them along to the next model in the chain.
+
+The nature of the modification is flexible and defined by the model. Some model plugins will calculate an impact metric (for example, energy from utilization) and add that to the observation before passing that to the next model. Some models will enrich the observation with meta-data required for other models, for example, adding grid carbon intensity values.
+
+## Initialization
+
+All the model plugins used in any component in the graph are configured at the top of the [Impl (Impact YAML)](Impl%20(Impact%20YAML).md) in the initialize section. There are multiple places to add configuration for model plugins; any configuration added here is global and applied to every use of the model throughout the graph.
+
+```yaml
+.
+.
+initialize:
+  plugins:
+    - name: <model-name-1>
+      path: <path-to-ts-module-to-load>
+      config:
+	       <key>: <value>
+    - name: <model-name-2>
+      path: <path-to-ts-module-to-load>
+      config:
+	       <key>: <value>	       
+.
+.
+graph: ~
+```
+
+## Usage
+
+In component nodes, we then configure the models we want to use in a pipeline like so:
+
+```yaml
+backend:
+  pipeline: "model-1 | model-2 | model-3"
+  config: 
+    model-1:
+      <key>: <value>
+    model-2:
+      <key>: <value>        
+    model-3:
+      <key>: <value>                
+  observations: 
+    - timestamp: 2023-07-06T00:00
+      duration: 5
+      cpu: 33
+    - timestamp: 2023-07-06T00:05
+      duration: 5
+      cpu: 23
+    - timestamp: 2023-07-06T00:10
+      duration: 5
+      cpu: 11
+```
+
+
+- `pipeline` defines the models we apply and the order in which we use them.
+- `config` in this part of the graph is config for each model in the pipeline (if any is required). Since we have multiple models, we need to define each config independently.
+- `observations` are the source observations, the values we are pumping into the start of this pipeline.
+
+The above is the short form; the long form would be to use a YAML array, like so:
+
+```yaml
+backend:
+  pipeline: 

```

Comment:
it will be better to maintain a single format.

Reply:
Ok good point, will update to one format